src/app/blog/healing-connections/page.mdx

```diff
@@ -0,0 +1,101 @@
+    description: post.description,
+    openGraph: {
+      title: post.title,
+      description: post.description,
+      images: [{
+        url: `/api/og?title=Blog&subtitle=${post.title}`,
+        width: 1200,
+        height: 630,
+        alt: post.title,
+        type: 'image/png',
+      }],
+      type: 'article'
+    }
+}
+
+export default (props) => <BlogPostLayout article={post} {...props} />
+
+We've put alot of effort into ensure that iroh connections "just work" in most situations. To achieve this, iroh’s networking stack does a lot of clever things. It allows you to dial by NodeID, using [certain discovery mechanisms](https://iroh.computer/blog/iroh-dns). It allows you to communicate with another machine, even if your computer and the other computer are each behind a NAT, using hole punching. 
+
+But did you know that iroh can maintain connections even if your network changes? Switch Wi-Fi networks, jump on a mobile hotspot, or turn on your VPN, and your connections stay intact. This works because connections correlate with your Node ID, not your IP address.
+
+## Healing Connections when you have a relay
+
+Iroh uses relay servers to coordinate hole-punching efforts and to heal connections when your network configuration changes. This process involves a few steps: 
+* detecting network interface changes,
+* sending probes,
+* using discovery messages to communicate those changes to your node. 
+
+The process completes when new addresses are added to our node’s address book. 
+
+### Detecting Network Configuration Changes
+
+When you create an iroh [`Endpoint`](https://docs.rs/iroh/latest/iroh/net/struct.Endpoint.html), you also start a **network monitoring** service that periodically checks the network interfaces and the routing table of your own computer. This is very *finicky* work—each OS has a different way of maintaining its routing table. We've supported Windows, Linux, and macOS ever since launching hole punching in iroh, and recently added support for [FreeBSD, NetBSD, and OpenBSD](https://github.com/n0-computer/iroh/pull/2311).
+
+If your network changes, like turning off mobile data to rely solely on Wi-Fi, these changes will show in the available interfaces. The network monitoring service will alert the iroh **`Endpoint`** to re-launch the next step: **netcheck**.
+
+### Netcheck
+
+Once we know the interfaces, we must determine what the outside world sees as our IP address and router configuration details. We do this by launching our **`netcheck`** probe, which sends STUN and ICMP probes to known relay servers to learn the node's public addresses and latency to the server.
+
+Once the probes finish, we have a list of addresses we can potentially be dialed on.
+
+Even if local endpoints change, we're still connected to any known relay nodes, maintaining a relayed connection. This allows us to send and receive data from the remote peer, albeit with potentially lower throughput and higher latency.
```

Comment:
Noooo... not "local endpoints".  let's ban that phrase!  You just used "public addresses" above.  Maybe you should re-use that.
```suggestion
Even if the public addresses change, we're still connected to any known relay nodes, maintaining a relayed connection. This allows us to send and receive data from the remote peer, albeit with potentially lower throughput and higher latency.
```

Reply:
it's been banned!!!!! 😂